resources/model/OpenStudio.idd

```diff
@@ -25576,47 +25576,145 @@ OS:Sizing:Zone,
+       \type choice
+       \key SupplyAirHumidityRatio
+       \key HumidityRatioDifference
+       \default HumidityRatioDifference
+  N19, \field Zone Dehumidification Design Supply Air Humidity Ratio
+       \note Zone Dehumidification Design Supply Air Humidity Ratio is only used when Zone Latent
+       \note Cooling Design Supply Air Humidity Ratio Input Method = SupplyAirHumidityRatio.
+       \note This input must be less than the zone humidity ratio at the
+       \note humidistat set point so that dehumidification can occur.
+       \minimum> 0.0
+       \type real
+       \units kgWater/kgDryAir
+  N20, \field Zone Cooling Design Supply Air Humidity Ratio Difference
+       \note Zone Dehumidification Design Supply Air Humidity Ratio Difference is only used when
+       \note Zone Latent Cooling Design Supply Air Humidity Ratio Input Method = HumidityRatioDifference.
+       \note This input is a positive value and defines the difference between the zone humidity
+       \note ratio at the thermostat and humidistat set point condition and the supply air
+       \note humidity ratio entering the zone.
+       \minimum> 0.0
+       \type real
+       \default 0.005
+       \units kgWater/kgDryAir
+  A10, \field Zone Latent Heating Design Supply Air Humidity Ratio Input Method
+       \note Use SupplyAirHumidityRatio to enter the humidity ratio when zone humidification
+       \note is required. The supply air humidity ratio should be greater than the zone humidity
+       \note ratio at the zone thermostat and humidistat set point condition.
+       \note Use HumidityRatioDifference to enter the difference in humidity ratio from the
+       \note zone thermostat and humidistat set point condition.
+       \type choice
+       \key SupplyAirHumidityRatio
+       \key HumidityRatioDifference
+       \default HumidityRatioDifference
+  N21, \field Zone Humidification Design Supply Air Humidity Ratio
+       \note Zone Humidification Design Supply Air Humidity Ratio is only used when Zone Latent
+       \note Heating Design Supply Air Humidity Ratio Input Method = SupplyAirHumidityRatio.
+       \note This input must be greater than the zone humidity ratio at the
+       \note humidistat set point so that humidification can occur.
+       \minimum> 0.0
+       \type real
+       \units kgWater/kgDryAir
+  N22, \field Zone Humidification Design Supply Air Humidity Ratio Difference
+       \note Zone Humidification Design Supply Air Humidity Ratio is only used when Zone Latent
+       \note Heating Design Supply Air Humidity Ratio Input Method = HumidityRatioDifference.
+       \note This input is a positive value and defines the difference between the zone humidity
+       \note ratio at the thermostat and humidistat set point condition and the supply air
+       \note humidity ratio entering the zone.
+       \minimum 0.0
+       \type real
+       \default 0.005
+       \units kgWater/kgDryAir
```

Comment:
@kbenne could you please take a look at this? These are new fields, likely not useful for most, so I left them with defaults and co instead of making them `\required-field` so they don't even show up in most cases when you serialize.

Could you advise whether you still would do them as `required-field` non optional, with default set in Ctor? Thanks!

Reply:
I do personally prefer required-fields, but I see your point. I think the PR is fine the way it is.

Reply:
Now I feel bad and kinda want to make them required...

Reply:
I made them required in f5eccf06dccdc0ebb965e4568ba102d7de4ae709, except the ones without a default: E+ doesn't provide a default for `Zone Dehumidification Design Supply Air Humidity Ratio` (and Humidification counterpart) and I really don't have a good default either, so leaving this as optional